yt/frontends/exodus_ii/data_structures.py

```diff
@@ -314,10 +314,12 @@ def _read_coordinates(self):
         with self._handle.open_ds() as ds:
             if "coord" not in ds.variables:
                 coords = np.array([ds.variables["coord%s" % ax][:]
-                                   for ax in coord_axes]).transpose().copy()
+                                   for ax in coord_axes]
+                                  ).transpose().astype(np.float64)
```

Comment:
Great catch.  One thing I would suggest is using the string "f8" to be consistent with other places in yt, but this is not a necessary change.